dimscord/restapi.nim

```diff
@@ -523,21 +470,15 @@ proc editGuildRole*(api: RestApi, guild_id: string, role_id: string;
             hoist: Option[bool] = none(bool);
             mentionable: Option[bool] = none(bool)): Future[Role] {.async.} =
     ## Modifies a guild role.
-    let payload = newJObject()
-    if name.isSome:
-        payload["name"] = %name
-    if color.isSome:
-        payload["color"] = %color
-    if hoist.isSome:
-        payload["hoist"] = %hoist
-    if mentionable.isSome:
-        payload["mentionable"] = %mentionable
+    let payload = %*{}
+
+    payload.loadOpt(name, color, hoist, mentionable)
     if pobj.isSome:
         payload["permissions"] = %(+(get(pobj)))
 
     result = (await api.request("POST", endpointGuildRoles(guild_id), $(payload))).newRole
 
-proc editGuildRolePosition*(api: RestApi, guild_id: string, role_id: string; position: Option[int] = none(int); reason: string = ""): Future[seq[Role]] {.async.} =
+proc editGuildRolePosition*(api: RestApi, guild_id: string, role_id: string, position: int, reason: string = ""): Future[seq[Role]] {.async.} =

```

Comment:
The optional position is needed. (see: https://discordapp.com/developers/docs/resources/guild#modify-guild-role-positions)